src/simulation/simulation_impl/world/nphysics_world.rs

```diff
@@ -722,6 +799,103 @@ mod tests {
         assert_eq!(Some(expected_object), actual_body);
     }
 
+    #[test]
+    fn passable_body_can_move_out_of_passable_body() {
+        test_body_can_move_out_of_passable_body(passable_body(), Point { x: 7.0, y: 7.0 });
+    }
+
+    #[test]
+    fn non_passable_body_can_move_out_of_passable_body() {
+        test_body_can_move_out_of_passable_body(physical_body(), Point { x: 7.0, y: 7.0 });
+    }
+
+    fn test_body_can_move_out_of_passable_body(
+        moving_body: PhysicalBody,
+        expected_location_after_moving: Point,
+    ) {
+        let rotation_translator = NphysicsRotationTranslatorImpl::default();
+        let mut world = NphysicsWorld::with_timestep(DEFAULT_TIMESTEP, box rotation_translator);
+
+        let moving_body_handle = world.add_body(moving_body.clone());
+
+        let passable_body = passable_body();
+        let passable_body_handle = world.add_body(passable_body.clone());
+
+        let force = Force {
+            torque: Torque::default(),
+            linear: Vector { x: 40.0, y: 40.0 },
+        };
+
+        world.apply_force(moving_body_handle, force);
+        world.step();
+
+        assert_eq!(
+            expected_location_after_moving,
+            world.body(moving_body_handle).unwrap().location
+        );
+
+        assert_eq!(
+            passable_body.location,
+            world.body(passable_body_handle).unwrap().location
+        );
+    }
+
+    #[test]
+    fn non_passable_body_can_move_into_passable_body() {
+        test_body_can_move_into_passable_body(
+            PhysicalBody {
+                location: Point { x: 15.0, y: 5.0 },
+                ..physical_body()
+            },
+            Point { x: 13.0, y: 5.0 },
+        );
+    }
+
+    #[test]
+    fn passable_body_can_move_into_other_passable_body() {
+        test_body_can_move_into_passable_body(
+            PhysicalBody {
+                location: Point { x: 15.0, y: 5.0 },
+                ..passable_body()
+            },
+            Point { x: 13.0, y: 5.0 },
+        );
+    }
+
+    fn test_body_can_move_into_passable_body(
```

Comment:
remove the `test_` prefix

Reply:
This is intentional, as this is not directly a test function, but called from functions marked as `#[test]`